ci-best-practices.md

```diff
@@ -0,0 +1,42 @@
+### CI should...
+
+- Be "as a service".
+- Push 1 or more check to PR's in Github.
+- Build on clean environments.
+- Build feature branch `pushes` against any upstream changes in deployable branches (staging/master).
+- Always use the "best practice project setup" `make` targets.
+- Notify "some channels" (Slack, email tbc) of a success or failure of a build.
+- Always keep built production artifacts in S3 for a minimum of X days.
+- Automatically deploy non feature branches (staging/master) when building.
+- Always have access to any off box resources it needs (e.g. AWS CLI).
+- Be able to run containered tests (tests must be fixtured).
+- Not allow sensitive information in scripts/configuration files.
+- Be able to run any build on any "agent/minion".
+- Allow us to pay for more concurrency.
+- Be able to cancel and rerun builds.
+- Give confidence that the code is shippable.
+- Allow stats to be gathered programmatically around our KPI's.
+
+### Potential make targets
```

Comment:
I think this need more padding out, but does not prevent this being merged. Just needs more changes in a future PR